client/src/middlewares/messageMiddleware.js

```diff
@@ -0,0 +1,34 @@
+import { webSocketActions, messageActions, userActions } from '../store/actions';
+
+const messageMiddleware = store => next => action => {
+    if (action.type === webSocketActions.types.received) {
+        const parsedMessage = JSON.parse(action.payload);
+        const userId = store.getState().user.id;
+        const actionToDispatch = getActionToDispatch(parsedMessage, userId);
+        if (actionToDispatch) {
+            return store.dispatch(actionToDispatch);
+        }
+    }
+    return next(action);
+};
+
+const getActionToDispatch = (parsedMessage, userId) => {
+    switch (parsedMessage.from.id) {
+        case 'server':
+            return parseServerMessage(parsedMessage);
+        case userId:
+            return;
```

Comment:
if you wont tell me, I would never know you got here

Reply:
yes this class is a bit in the making, but what exactly is the problem?
I thought about putting the 'server' in a config but it's a bit of a OverKill and spam the config file maybe a const is more clean?